tfx/utils/path_utils.py

```diff
@@ -28,42 +29,36 @@
 EVAL_MODEL_DIR = 'eval_model_dir'
 SERVING_MODEL_DIR = 'serving_model_dir'
 
-# TODO(b/127149760): simplify this PPP-esque structure.
-#
-# Directory structure of exported model for estimator based trainer:
-#   |-- <ModelExportPath>
-#       |-- EVAL_MODEL_DIR  <- eval_model_dir
-#           |-- <timestamped model>  <- eval_model_path
-#               |-- saved_model.pb
-#               |-- ...
-#       |-- SERVING_MODEL_DIR  <- serving_model_dir
-#           |-- export
-#               |-- <exporter name>
-#                   |-- <timestamped model>  <- serving_model_path
-#                       |-- saved_model.pb
-#                       |-- ...
-#           |-- ...
-#
-# For generic trainer with Keras, there won't be eval model:
-#   |-- <ModelExportPath>
-#       |-- SERVING_MODEL_DIR  <- serving_model_dir
-#           |-- saved_model.pb
-#           |-- ...
+"""
+Directory structure of exported model for estimator based trainer:
+  |-- <ModelExportPath>
+      |-- EVAL_MODEL_DIR  <- eval_model_dir, eval_model_path
+          |-- saved_model.pb
+          |-- ...
+      |-- SERVING_MODEL_DIR  <- serving_model_dir, serving_model_path
+          |-- saved_model.pb
+          |-- ...
 
+For generic trainer with Keras, there won't be eval model:
+  |-- <ModelExportPath>
+      |-- SERVING_MODEL_DIR  <- serving_model_dir, serving_model_path
+          |-- saved_model.pb
+          |-- ...
+"""
 
 def eval_model_dir(output_uri: Text) -> Text:
   """Returns directory for exported model for evaluation purpose."""
   return os.path.join(output_uri, EVAL_MODEL_DIR)
 
 
 def eval_model_path(output_uri: Text) -> Text:
-  """Returns path to timestamped exported model for evaluation purpose."""
+  """Returns final path to exported model for evaluation purpose."""
```

Comment:
merge eval_model_working_path into this eval_model_path (will be `if` `elif` + `else`)

as downstream still call eval_model_path, we want downstream to work with old version (e.g., a model artifact created by previous tfx version is resolved by Resolver to use in current pipeline)

Reply:
since eval model is timestamped, i've just replaced it with a try catch (i.e. try to get path and except if it doesnt exist).

Reply:
you can check if saved_model.pb exists in model_dir, if so return model_dir else return io_utils.get_only_uri_in_dir(model_dir)